backend/cfg/cfg_polling.ml

```diff
@@ -188,21 +149,54 @@ let potentially_recursive_tailcall :
     Misc.fatal_error
       "Cfg_polling.potentially_recursive_tailcall has been interrupted"
 
+(* Insertion of poll instruction onto back edges: a poll instruction is needed
+   if there is an unsafe path from the header of the loop to the back edge. An
+   unsafe path is simply a path such that all blocks are unsafe. *)
+
+let exists_unsafe_path :
+    Cfg.t -> safe_map:bool Label.Tbl.t -> from:Label.t -> to_:Label.t -> bool =
+ fun cfg ~safe_map ~from ~to_ ->
+  let exception Found in
+  try
+    let open_ = ref (Label.Set.singleton from) in
+    let closed = ref Label.Set.empty in
+    while not (Label.Set.is_empty !open_) do
+      let label = Label.Set.choose !open_ in
+      if Label.equal label to_ then raise Found;
```

Comment:
I think the current implementation is overly conservative, we should check that `label` is not safe before `raise Found`. 
For example: H->B and B->H where A is the loop header block, A it is not safe, B is safe. The result of `exists_unsafe_path ~from:H ~to_:B` will be true, I think. It's not taking into account safe instructions in `to_` block.

Reply:
As discussed off-line, we will address the
issue in a follow-up pull request.

Reply:
(Should be addressed by https://github.com/ocaml-flambda/flambda-backend/pull/3113.)